common/src/chain/transaction/input.rs

```diff
@@ -1,22 +1,29 @@
+use crate::chain::block::Block;
 use crate::chain::transaction::Transaction;
 use crate::primitives::Id;
 use parity_scale_codec::{Decode, Encode};
 
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum OutpointSource {
+    Transaction(Transaction),
+    Block(Block),
+}
+
 #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub struct OutPoint {
-    id: Id<Transaction>,
+    id: Id<OutpointSource>,
```

Comment:
This is not correct. The Idable type is Transaction or Block. Not the enum. So the enum should be:

```
pub enum OutpointSource {
    Transaction(Id<Transaction>),
    Block(Id<Block>),
}
```